hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/util/TestExitUtil.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.util;
+
+import static org.junit.Assert.*;
+
+import org.apache.hadoop.util.ExitUtil.ExitException;
+import org.apache.hadoop.util.ExitUtil.HaltException;
+
+import org.junit.Test;
+
+public class TestExitUtil {
+
+  @Test
+  public void testGetSetExitExceptions() {
+    // prepare states and exceptions
+    ExitUtil.disableSystemExit();
+    ExitUtil.resetFirstExitException();
+    ExitException ee1 = new ExitException(1, "TestExitUtil forged 1st ExitException");
+    ExitException ee2 = new ExitException(2, "TestExitUtil forged 2nd ExitException");
+    try {
+      // check proper initial settings
+      assertFalse(ExitUtil.terminateCalled());
+      assertNull(ExitUtil.getFirstExitException());
+
+      // simulate/check 1st call
+      try {
+        ExitUtil.terminate(ee1);
```

Comment:
use LambdaTestUtils.intercept() to make the call; and we can use assertSame for equivalenced
```
ExitException ex = intercept(ExitException.class, () -> 
  ExitUtil.terminate(ee1));
assertSame("ExitUtil.terminate should have rethrown its ExitException argument but it "
            +"had thrown something else", ee1, ee);
```

Reply:
Ok. I've checked how other tests using intercept were coded. So now the test methods all declare to throw Throwable like them. Plus all the assertEquals are now assertSame ones.